File: javascript-sources/chapter-1/demo-01/query-dynamodb.js
// Load the SDK for JavaScript
let AWS = require('aws-sdk');

let common = require('./common');
common.configure_credentials(AWS);

let docClient = new AWS.DynamoDB.DocumentClient();

let query_input = {
    TableName : "orders",
    KeyConditionExpression: "#id = :id and begins_with(#ts, :ts)",
    ExpressionAttributeNames: {
        "#id": "user_id",
        "#ts":"order_time"
    },
    ExpressionAttributeValues: {
        ":id":"001",
        ":ts":"2019-0"
    }
};

// Call DynamoDB to add the item to the table
docClient.query(query_input, function(err, data) {
    if (err) {
        console.log("Error", err);
    } else {
        console.log(data.Items);
    }
});

File: python-sources/chapter-1/demo-01/query_dynamodb.py
import boto3
from boto3.dynamodb.conditions import Key, Attr

session = boto3.Session(profile_name='admin')

dynamodb = session.resource("dynamodb", region_name="us-west-2")
table = dynamodb.Table('orders')

response = table.query(
    KeyConditionExpression=Key('user_id').eq('101') & Key('order_time').begins_with('2019')
)

print(response['Items'])